src/jvmTest/kotlin/com/sksamuel/kotest/tests/ktor/KtorAssertionsTest.kt

```diff
@@ -58,6 +60,20 @@ class KtorAssertionsTest : StringSpec({
       }
    }
 
+   "test headers matcher fail" {
+      withTestApplication({ testableModule() }) {
+         handleRequest(HttpMethod.Get, "/").apply {
+            shouldThrow<AssertionError> {
+               response.shouldHaveHeader("fail_name", "fail_value")
+            }.message shouldBe "Response should have header fail_name=fail_value but fail_name=null"
```

Comment:
I referred  [kotest](https://github.com/kotest/kotest/blob/015fcd574ec4421d4e4398d8c88dd145f1ea8339/kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/boolean/BooleanMatchersTest.kt#L38-L46) to create these test cases.